e2e/datasource/datasource_test.go

```diff
@@ -0,0 +1,257 @@
+package integration_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/grafana-labs/surrealdb-datasource/pkg/client"
+	"github.com/grafana-labs/surrealdb-datasource/pkg/plugin"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+)
+
+// table is a list of test cases for the QueryData method.
+var cases = []struct {
+	input string
+	name  string
+}{
+	{
+		input: "SELECT * FROM person WHERE name = 'Richie Buck';",
+		name:  "SELECT query with WHERE clause",
+	},
+	{
+		input: "SELECT id, name FROM person;",
+		name:  "SELECT specific columns",
+	},
+	{
+		input: "SELECT count() FROM person GROUP all;",
+		name:  "Aggregate function",
+	},
+	{
+		input: "SELECT * FROM person ORDER BY name DESC;",
+		name:  "SELECT query with ORDER BY clause",
+	},
+	{
+		input: "SELECT * FROM person WHERE first_name ~ 'rich';",
+		name:  "SELECT with fuzzy equality",
+	},
+	{
+		input: "SELECT * FROM person WHERE company_name == NONE;",
+		name:  "SELECT with IS NONE",
+	},
+	{
+		input: "SELECT * FROM person WHERE company_name != NONE;",
+		name:  "SELECT with IS NOT NONE",
+	},
+	{
+		input: "SELECT * FROM person LIMIT 10;",
+		name:  "SELECT with LIMIT",
+	},
+	{
+		input: "SELECT product.name FROM review;",
+		name:  "SELECT with Record links",
+	},
+}
+
+// getSurrealEndpoint returns the endpoint for the SurrealDB server.
+func getSurrealEndpoint() string {
+	env := os.Getenv("SURREAL_DB_URL")
+
+	if env != "" {
+		return env
+	}
+
+	if os.Getenv("CI") != "true" {
+		return "ws://localhost:8000/rpc"
+	}
+
+	return "ws://surrealdb:8000/rpc"
+}
+
+// createTestInstance creates a new SurrealDatasource instance for testing.
+func createTestInstance() (instancemgmt.Instance, error) {
+	config := client.SurrealConfig{
+		Database:  "test",
+		Endpoint:  getSurrealEndpoint(),
+		Namespace: "test",
+		Username:  "root",
+	}
+
+	msg, err := json.Marshal(config)
+
+	if err != nil {
+		return nil, err
+	}
+
+	dsiConfig := backend.DataSourceInstanceSettings{
+		JSONData:                msg,
+		DecryptedSecureJSONData: map[string]string{"password": "test"},
+	}
+
+	return plugin.NewDatasource(context.Background(), dsiConfig)
+}
+
+// createJsonRequest creates a new JSON request for testing.
+func createJsonRequest(query string) json.RawMessage {
+	return json.RawMessage(`{"queryText":"` + query + `"}`)
+}
+
+func TestNewDatasource(t *testing.T) {
+	instance, err := createTestInstance()
+
+	if err != nil {
+		t.Errorf("unexpected error: %s", err)
```

Comment:
The only nit from me is to not use conditions to cause the tests to error but instead use the `assert` and `require` testing functions. Wdyt?

Reply:
Those require `testify` though right?

Reply:
Yep, feel free to ignore the suggestion I just think it'll improve the readability 😊